### ISSUE TITLE
Finetuning for semi-transparent glass background removal

### ISSUE THREAD

**Issue by mehamednews** [ZhengPeng7/BiRefNet#42]
Hi,
First, I'd like to thank you for your amazing work (& the amount of effort you're putting into answer peoples questions).
I'd appreciate your insights & suggestions regarding the **possibility** of fine-tuning one of the pre-trained models (not sure which one would be best here) on images of cars.
My main goal is to remove what's behind the windows of cars. I think an example will explain this better:
![100307](https://github.com/user-attachments/assets/4df059af-c6f1-4a1b-b27c-314cdc1c5c39)
![100307-mask](https://github.com/user-attachments/assets/957ef814-2b15-4992-bf5f-aeddacc1696a)

**Comment by ZhengPeng7** [comment]
Do you mean you want to extract the target car by a mask with a transparent channel?

Regarding the dataset, ~50k images are enough for training, in my experience, even from scratch. Perhaps training from scratch could be better (I'm not sure) -- 1). ~50k is already a large number, only if the mask labels are valid and sound; 2). images were resized to 1024x1024 in my experiments, so there might be a gap between your images (800x600).

But anyway, only experiments can tell everything. Tell me if you have any problem with it.

**Comment by mehamednews** [comment]
Thanks for the quick answer.
Yes, I check the `DIS` dataset and I noticed they have `solid` masks (either 100% white or 100% black)
what I'm trying to achieve is to get gray areas in areas corresponding to windows where the environment peaks through (not sure how to explain it :smile:)
![image](https://github.com/user-attachments/assets/b13b10c6-a280-4b79-b434-3ecc3d0e1364)

I'm going to create a dataset with `1024px` in width (still 4:3) and test with it.

**Comment by ZhengPeng7** [comment]
Thanks for your explanation. I understand your need here, where the data is similar to that in the matting tasks. For example, here is a GT example below from the portrait segmentation dataset [P3M-10k](https://github.com/JizhiziLi/P3M) -- see the pixel values in the regions of hair for an easy check.

One more important thing is that though GT labels in datasets like DIS5K are in {0, 1}, the predicted maps are always float numbers in the range of (0, 1), some techniques were even proposed to push these values more confident -- to 0 or to 1, instead of to 0.5.
What I want to mean is that it only depends on the datasets provided for training. And as I said before, there might be data domain gap between your custom data and DIS5K or used data. Therefore, if you have insufficient GPUs, you can fine-tune the provided general-use weights. If possible, I recommend training from scratch to examine the accuracy.

![p_0b5f94f1](https://github.com/user-attachments/assets/93db390e-5526-4b24-bf95-92419333dd30)

**Comment by akansal1** [comment]
hey @mehamednews were you able to get the desired results also how did you generate this data ? 

@ZhengPeng7 is their a dataset that has transparency related samples, i want to understand how transparency works, I am struggling to find if Glass transparency is background removal problem

**Comment by ZhengPeng7** [comment]
Only if the GTs have transparency will the model learn it. It depends on your practical use cases to decide whether to keep it.

**Comment by mehamednews** [comment]
@akansal1 I generated the images & the masks using blender (took me sometime to get a realistic look of cars)
I wasn't able to get perfect results by fine tuning the model but they're close (my dataset wasn't perfect I guess)
Here's are some examples:
![image](https://github.com/user-attachments/assets/196b43dd-ff4c-4356-acca-72f54725d661)
![image](https://github.com/user-attachments/assets/32896cf2-d972-4a88-a0aa-a8f3ffbe2d93)
![image](https://github.com/user-attachments/assets/0da1c69b-dc74-45b3-9507-b1b6f72d664e)
the third image isn't as good.
I think with more data and a lower learning rate (maybe even training from scratch - which is not what I did) would give much better results.

as always, thanks a lot @ZhengPeng7 for the amazing work.

**Comment by akansal1** [comment]
thank you @ZhengPeng7 and @mehamednews and no doubt great work :)

**Comment by ZhengPeng7** [comment]
Thanks a lot for your sharing, @mehamednews; this is very valuable!
